### ISSUE TITLE
Reference documentation doesn't list extended classes and implemented interfaces

### ISSUE THREAD

**Issue by jnizet** [angular/angular#10939]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

First, I'm sorry if I'm submitting this to the wrong project. This bug is about the reference documentation, and I don't know if I should submit it here or in angular.io or somewhere else. Please forgive me if it's the wrong place.

**Current behavior**

The reference documentation doesn't mention the superclass and implemented interfaces of the classes, making it impossible to find the attributes and methods actually available on a type, and to understand the examples listed in the reference doc or in the tutorials, without reading the source code.

For example, the documentation for NgForm has an example showing how to use its `value` attribute to submit the form, but there is no `value` attribute in NgForm. The only way I could find this information was by reading the source code and discovering that value was a field of the inherited `AsbtractControlDirective` class. 

**Expected/desired behavior**

The reference documentation should, at minimum, display the superclass and implemented interfaces of all classes.

A link would be even better (and same for the types of the attributes, arguments and return types, by the way, which could link to the documentation of their type)

Even better would be to have a list of inherited members, linking directly to their documentation in the superclass documentation.

In short, I expect the same level of usability as the javadoc, for example.

**Reproduction of the problem**
Go to https://angular.io/docs/ts/latest/api/forms/index/NgForm-directive.html, and try to find if NgForm really has a `value` or a `valid` attribute.

**What is the expected behavior?**
I should be able to find this information without having to read the source code.

**What is the motivation / use case for changing the behavior?**
Understand the examples. Being able to discover attributes and methods without reading the source code

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.5
 
* **Browser:** all 
 
* **Language:** TypeScript

**Comment by epelc** [comment]
Is this the same issue as the docs not showing eg `HostBindingDecorator` docs for `HostBinding`?

Notice the docs site is near empty with only the `HostBinding` info while there is a bunch of extra info if you lookup `HostBindingDecorator`. 
![image](https://cloud.githubusercontent.com/assets/5204642/19630488/6dcdd9d8-9959-11e6-8e30-554830512ff5.png)

I noticed this because atom-typescript finds the docs for `HostBindingDecorator` when you use `HostBinding`. This also happens for `HostListener`.

**Comment by vicb** [comment]
> First, I'm sorry if I'm submitting this to the wrong project. This bug is about the reference documentation, and I don't know if I should submit it here or in angular.io or somewhere else. Please forgive me if it's the wrong place.

Please report docs issue on angular/angular.io
